src/main/java/graphql/schema/PropertyDataFetcher.java

```diff
@@ -175,13 +177,47 @@ private Object getPropertyViaGetterUsingPrefix(Object object, String prefix, Met
 
     @SuppressWarnings("SimplifiableIfStatement")
     private boolean isBooleanProperty(GraphQLOutputType outputType) {
-        if (outputType == GraphQLBoolean) return true;
+        if (outputType == GraphQLBoolean) {
+            return true;
+        }
         if (outputType instanceof GraphQLNonNull) {
             return ((GraphQLNonNull) outputType).getWrappedType() == GraphQLBoolean;
         }
         return false;
     }
 
+    private static final ConcurrentMap<String, Method> METHOD_CACHE = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<String, Field> FIELD_CACHE = new ConcurrentHashMap<>();
+
+    /**
+     * PropertyDataFetcher caches the methods and fields that map from a class -> a property for runtime performance reasons.
```

Comment:
javadoc fails on using `->` in a comment, change it to `to`

Reply:
fixed